main.tf

```diff
@@ -656,6 +656,68 @@ resource "helm_release" "castai_kvisor_self_managed" {
   }
 }
 
+#---------------------------------------------------#
+# CAST.AI Workload Autoscaler configuration         #
+#---------------------------------------------------#
+resource "helm_release" "castai_workload_autoscaler" {
+  count = var.install_workload_autoscaler && !var.self_managed ? 1 : 0
+
+  name             = "castai-workload-autoscaler"
+  repository       = "https://castai.github.io/helm-charts"
+  chart            = "castai-workload-autoscaler"
+  namespace        = "castai-agent"
+  create_namespace = true
+  cleanup_on_fail  = true
+  wait             = true
+
+  version = var.workload_autoscaler_version
+  values  = var.workload_autoscaler_values
+
+  set {
+    name  = "castai.apiKeySecretRef"
+    value = "castai-cluster-controller"
+  }
+
+  set {
+    name  = "castai.configMapRef"
+    value = "castai-cluster-controller"
+  }
+
+  depends_on = [helm_release.castai_agent]
+
+  lifecycle {
+    ignore_changes = [version]
+  }
+}
+
+resource "helm_release" "castai_workload_autoscaler_self_managed" {
```

Comment:
why this is needed?

Reply:
same thought but looks like main.tf has all of our components like this..

Reply:
self_managed mode allow users to control the version and deployment of the component directly through Terraform

Reply:
@varnastadeus Is it ok to proceed with the PR?

Reply:
@varnastadeus are you fine with merge of this?

Reply:
I would also vote for merge and release if it's possible 🙋‍♂️